dev/scripts/src/alluxio.org/check-docs/main.go

```diff
@@ -257,6 +275,108 @@ func getSingleRegexMatch(re *regexp.Regexp, l string) (string, error) {
 	return namedMatch, nil
 }
 
+type Subfile struct {
+	Title string `yaml:"title"`
+	URL   string `yaml:"url"`
+}
+
+type File struct {
+	ButtonTitle string    `yaml:"buttonTitle"`
+	Subfiles    []Subfile `yaml:"subfiles"`
+	Subitems    []File    `yaml:"subitems"`
+}
+
+// get url from menuPath with two checks one is the buttonTitle check and the second is the url end checks
+func parseMenuUrl(menuPath string) (map[string]struct{}, error) {
+	content, err := ioutil.ReadFile(menuPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file at %v with message : \n %v", menuPath, err)
+	}
+	var ret []File
+	if err := yaml.Unmarshal(content, &ret); err != nil {
+		return nil, fmt.Errorf("error unmarshalling config from:\n%v with message : \n %v", string(content), err)
+	}
+	menuMap := map[string]struct{}{}
+	var errMsgs []string
+	checkAndSaveUrl(ret, &menuMap, errMsgs)
+	if len(errMsgs) > 0 {
+		return nil, fmt.Errorf("encountered errors parsing %v:\n%v", menuPath, strings.Join(errMsgs, "\n"))
+	}
+	return menuMap, nil
+}
+
+// recursion function for more levels of docs
+func checkAndSaveUrl(files []File, menuMap *map[string]struct{}, errMsgs []string) {
```

Comment:
```suggestion
func checkAndSaveUrl(files []File, menuMap map[string]struct{}, errMsgs []string) {
```
maps and slices are pointers, so you can directly pass them into functions as arguments without making a pointer reference